### PR TITLE
Update release.yml

### PR DESCRIPTION
**Pull request by Flamefire**
- Avoid Copy of all files
- Prune all git and CI (meta) data files and folders
- Convert line endings to Windows/Dos for the zip file

Stripped down example run: https://github.com/Flamefire/locale/actions/runs/3481281759/jobs/5822170011

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -12,20 +12,42 @@ jobs:
...
-          cd ..
-          cp -a ${{ github.event.repository.name }} ${{ github.ref_name }}
-          rm -rf ${{ github.ref_name }}/.git
-          zip -q ${{ github.ref_name }}.zip -r ${{ github.ref_name }}
           tar -czf ${{ github.ref_name }}.tar.gz ${{ github.ref_name }}
           tar -cJf ${{ github.ref_name }}.tar.xz ${{ github.ref_name }}
+      - name: Create Windows archives
+        run: |
+          sudo apt-get -o Acquire::Retries=5 install -y dos2unix
+          find ${{ github.ref_name }} -type f -exec unix2dos {} \;
```
Comment (Lastique) [inline .github/workflows/release.yml:45]: I think this processes excess files. `.sh` files must retain UNIX line endings even on Windows, and you should not process any binary files, e.g. images in docs.

I still think the best way to do it is to build Windows archives on Windows. git will do the conversion on checkout for you.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -12,20 +12,42 @@ jobs:
...
           submodules: true
 
-      - name: Create archives
+      - name: Cleanup
+        shell: bash
+        run: |
+          # Remove all .git folders & files as well as .github folders and .git, .gitattribute etc. files
+          find ${{ github.ref_name }} -name ".git*" -prune -exec rm -rf {} +
+          # Remove CI configs
+          find ${{ github.ref_name }} \( \
```
Comment (Lastique) [inline .github/workflows/release.yml:24]: I'd add `-type f` as well.

File: .github/workflows/release.yml (hunk 1 above)
Comment (mojca) [reply .github/workflows/release.yml:45]: This certainly sounds way too dangerous.

Comment (Flamefire) [reply .github/workflows/release.yml]: unix2dos in recent version doesn't modify binary files as I've learned. But true on sh files, so added a windows job which makes it a bit more complicated and I hope the release uploader correctly finds the files and combines them into that one release

File: .github/workflows/release.yml (hunk 2 above)
Comment (Flamefire) [reply .github/workflows/release.yml]: Done

**Comment by Flamefire** [comment]
Test run with a [simplified config](https://github.com/Flamefire/locale/blob/test/.github/workflows/ci.yml) (no upload) is at https://github.com/Flamefire/locale/actions/runs/3482065247

**Comment by pdimov** [comment]
I'm not quite sure what will happen if the two jobs try to create the same release simultaneously.

**Comment by Flamefire** [comment]
I'm not sure either, haven't used that action yet and the issue list doesn't look to promising.
Previously I used actions/create-release & actions/upload-release-asset and the latter seems to work for uploading multiple assets.

We could split it into 2/3 jobs: 1 creating the release and 2 publishing the assets (1 each), see e.g. [Publish and upload](https://github.com/Return-To-The-Roots/s25client/blob/0537322eb59cd418697ceeb35f21564150f0bdf4/.github/workflows/release.yml#L61-L84) and [Concurrent upload](https://github.com/Return-To-The-Roots/libsiedler2/blob/eea3744814d347baaff2923e6b0572b6f9de08ae/.github/workflows/release.yml#L125-L134)

**Comment by pdimov** [comment]
> Previously I used actions/create-release & actions/upload-release-asset

Those were my first choice but both are deprecated.

**Comment by pdimov** [comment]
I wonder why `7z` works for you without supplying a command, `7z a`. It fails for me.

**Comment by Flamefire** [comment]
> I wonder why 7z works for you without supplying a command, 7z a. It fails for me.

It doesn't. Missed that somehow while copying the stuff from my test file

BTW: Instead of repeating the code you can still use the matrix approach from this PR and `maxparallel: 1`: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#defining-the-maximum-number-of-concurrent-jobs

**Comment by pdimov** [comment]
I don't mind repeating the code here, it's clearer this way and not that much of a maintenance burden as it fits on a screen. :-)

**Pull request closed without merging**